lib/pushmi_pullyu/cli.rb

```diff
@@ -160,15 +130,54 @@ def run_tick_loop
     end
   end
 
+  def running?
+    @running
+  end
+
+  def setup_log
+    if config.daemonize
+      PushmiPullyu::Logging.initialize_logger(config.logfile)
+    else
+      logger.formatter = PushmiPullyu::Logging::SimpleFormatter.new
+    end
+    logger.level = ::Logger::DEBUG if config.debug
+  end
+
+  def setup_signal_traps
+    Signal.trap('INT') { Thread.new { shutdown } }
+    Signal.trap('TERM') { Thread.new { shutdown } }
+    Signal.trap('HUP') { Thread.new { PushmiPullyu::Logging.reopen } }
```

Comment:
Complex question, but is this threadsafe? It's going to depend on the underlying logger, but if it's not, we could see some weirdness here. It looks like it could update @logger in one thread while another is writing through it, which could do some weird things (half of the logs from a single loop end up in one file, half in another, for one possibility). 

One way to deal with this might be to set a flag on the main run loop, and then reset the logger at the end or beginning of the main loop.

(Having flashbacks to CMPUT 379 right now)